### DISCUSSION TITLE
SUIT (Software Updates for Internet of Things)

### DISCUSSION THREAD

**Discussion by athoelke** [ARM-software/psa-firmware-update-spec#6] [General]
The IETF SUIT specifications are relevant for this API specification and the working group. This is a suite of emerging standards around Software update for IoT, and particularly MCU-based, systems.

SUIT defines an 'envelope' for describing the meta-data about the firmware update, that supports Secure update and Secure boot policies. It also contains instructions for the verification and installation of the update, requiring a small interpreter in the target device.

### Analysis

In a SUIT-based update, this envelope processor becomes the 'orchestrator' of the device update, following the instructions provided by the firmware creator in the SUIT envelope. For a secure update, this processing occurs inside the Root of Trust.

Placing the update orchestration inside the Root of Trust has implications for the API between the application and secure contexts. I expect that this will not be exactly the same as a FWU API that is effective for for Client-orchestrated updates, where the Update Service follows the Client's instructions, subject only to Service-side policy enforcement.

### Proposal

I would like to put "designing/optimizing for SUIT" out of scope for version 1.0 of the PSA Firmware Update API, to be addressed in a future update to the API.

### Further information about SUIT

SUIT architecture (RFC 9019): https://datatracker.ietf.org/doc/rfc9019/
Manifest information model (RFC 9124): https://datatracker.ietf.org/doc/rfc9124/

SUIT paper: https://hal.inria.fr/hal-02351794/document

IETF working group: https://datatracker.ietf.org/wg/suit/about/

**Comment by d3zd3z** [comment]
It looks like the initial approach we're going to be taking with regards to SUIT and MCUboot is going to be to have a pre-processor that processes the manifest, but instead of performing the actions, stores the data in memory, in a similar style to the current TLV. Then the existing processing could happen. Something like this could be done at the FWU level if SUIT ends up being needed before the API concerns could be addressed.

**Reply by athoelke** [reply]
I definitely want to work through the design options for implementing SUIT efficiently in a system where the network access (Update Client) is segregated from the installer logic (Update Service), and there might be more than one step in evolving a system to that point.

But, I would prefer to do that once it is clear that the other issues around this API can be addressed to the satisfaction of the working group for a stable v1.0 for DevSummit.
